tests/integration/query/json/with_aggregate_test.go

```diff
@@ -0,0 +1,55 @@
+package json
```

Comment:
todo: Please add some more tests.  At least one to cover the `default` case in the switch, and another where `isMapped` is false.  Also consider tests for invalid user filters, although that could arguably be done in the other JSON filter PR.

Reply:
+1

Reply:
I've added an additional test to cover the `isMapped` case. The `default` case is technically dead code but removing it could cause a hard to find bug in the future.

Reply:
Okay no worries, I have a very minor preference for an error to be returned from `default` then, but happy as is, so only change it if you like the suggestion.

Thanks for the `isMapped` test :)